corehq/apps/es/domains.py

```diff
@@ -55,9 +60,38 @@ class ElasticDomain(ElasticDocumentAdapter):
     def mapping(self):
         return get_adapter_mapping(self)
 
-    @classmethod
-    def from_python(cls, doc):
-        return from_dict_with_possible_id(doc)
+    def from_python(self, domain):
```

Comment:
given that this is basically the same in all adapters, I think it would make sense to break it out into a util function (maybe one that takes the expected class as a parameter), or even just on the base class and have the doc class part of the subclass

Reply:
That is a good suggestion. I thought about it when I was implementing it but given how this function is used which is majorly in index function, the API of an index would be changed to take in a class also which I guess would not be ideal. So I did not go that route.

Reply:
I don't quite follow your reasoning @AmitPhulera. If you made it a function you could just call the function here instead of having the full implementation.

If you did want to put it on the base class you could still call it from here with the correct type (or else add the type as a class level attribute so that it doesn't need to be passed in.

Reply:
https://github.com/dimagi/commcare-hq/pull/32660/commits/0b0c9914bf4a6affe39035ee23efef4eacd55456 I have moved the function to base class and subsequent commits updates the usage in adapters.